ml-agents/mlagents/trainers/models.py

```diff
@@ -15,6 +14,19 @@
 EPSILON = 1e-7
 
 
+class Tensor3DShape(NamedTuple):
+    height: int
+    width: int
+    num_channels: int
+
+    @property
+    def gray_scale(self) -> bool:
+        return self.num_channels == 1
+
+    def __str__(self):
```

Comment:
You get this for free with NamedTuple

Reply:
👍  I removed this and the `gray_scale` property since they no longer make sense.